docs/user/reference/environment-variables.rst

```diff
@@ -44,6 +44,15 @@ All :doc:`build processes </builds>` have the following environment variables au
 
     :Example: ``/home/docs/checkouts/readthedocs.org/user_builds/project/envs/version``
 
+.. envvar:: READTHEDOCS_CANONICAL_URL
+
+    Canonical URL for the version it's currently building.
+    If the project has configured a :doc:`custom domain </custom-domains>` (e.g. ``docs.example.com``) it will be used in the resulting canonical URL.
+    Otherwise, the default domain will be exposed (e.g. ``.readthedocs.io``)
```

Comment:
```suggestion
    Otherwise, your project's :ref:`default subdomain <custom-domains:default-subdomain>` will be used.
    
    The path for the language and version is appended to the domain, so the final canonical base URLs can look like this:
```

Reply:
```suggestion
    Otherwise, your project's :ref:`default subdomain <custom-domains:default-subdomain>` will be used.
    
The path for the language and version is appended to the domain, so the final canonical base URLs can look like the following examples.```